skopt/space.py

```diff
@@ -5,98 +5,70 @@
 from scipy.stats.distributions import rv_discrete
 from scipy.stats.distributions import uniform
 
-from sklearn.preprocessing import LabelBinarizer
 from sklearn.utils import check_random_state
 from sklearn.utils.fixes import sp_version
 
+from skopt.transformers import CategoricalEncoder
+from skopt.transformers import HyperCube
+from skopt.transformers import Identity
+from skopt.transformers import Log10
+from skopt.transformers import Pipeline
 
 # helper class to be able to print [1, ..., 4] instead of [1, '...', 4]
 class _Ellipsis:
     def __repr__(self):
         return '...'
 
 
-class _Identity:
-    """Identity transform."""
-
-    def fit(self, X):
-        return self
-
-    def transform(self, X):
-        return X
-
-    def inverse_transform(self, Xt):
-        return Xt
-
-
-class _Log10:
-    """Base 10 logarithm transform."""
-
-    def fit(self, X):
-        return self
-
-    def transform(self, X):
-        return np.log10(np.asarray(X, dtype=np.float))
-
-    def inverse_transform(self, Xt):
-        return 10.0 ** np.asarray(Xt, dtype=np.float)
-
-
-class _CategoricalEncoder:
-    """OneHotEncoder that can handle categorical variables."""
-
-    def __init__(self):
-        """Convert labeled categories into one-hot encoded features."""
-        self._lb = LabelBinarizer()
-
-    def fit(self, X):
-        """Fit a list or array of categories.
-
-        Parameters
-        ----------
-        * `X` [array-like, shape=(n_categories,)]:
-            List of categories.
-        """
-        self.mapping_ = {v: i for i, v in enumerate(X)}
-        self.inverse_mapping_ = {i: v for v, i in self.mapping_.items()}
-        self._lb.fit([self.mapping_[v] for v in X])
-        self.n_classes = len(self._lb.classes_)
-
-        return self
-
-    def transform(self, X):
-        """Transform an array of categories to a one-hot encoded representation.
-
-        Parameters
-        ----------
-        * `X` [array-like, shape=(n_samples,)]:
-            List of categories.
-
-        Returns
-        -------
-        * `Xt` [array-like, shape=(n_samples, n_categories)]:
-            The one-hot encoded categories.
-        """
-        return self._lb.transform([self.mapping_[v] for v in X])
-
-    def inverse_transform(self, Xt):
-        """Inverse transform one-hot encoded categories back to their original
-           representation.
-
-        Parameters
-        ----------
-        * `Xt` [array-like, shape=(n_samples, n_categories)]:
-            One-hot encoded categories.
-
-        Returns
-        -------
-        * `X` [array-like, shape=(n_samples,)]:
-            The original categories.
-        """
-        Xt = np.asarray(Xt)
-        return [
-            self.inverse_mapping_[i] for i in self._lb.inverse_transform(Xt)
-        ]
+def check_dimensions(dimensions, transform=None):
```

Comment:
Originally thought about having `check_dimension()` (no s) and the loop being performed by the caller. It would make it easier for the caller to decide what transform to they wanted for what dimension. Not sure if that is needed or over engineering. You decide.